tests/iscsi/iscsi_server.pm

```diff
@@ -11,96 +11,242 @@
+    send_key 'alt-t';    # select target field
+    wait_still_screen(stilltime => 1, timeout => 5, similarity_level => 44);
+    send_key 'ctrl-a';    # select all text inside target field
+    wait_still_screen(stilltime => 1, timeout => 5, similarity_level => 45);
+    send_key 'delete';    # text it is automatically selected after tab, delete
+    type_string_slow_extended $test_data->{target_conf}->{name};
+    send_key 'tab';       # tab to identifier field
+    wait_still_screen(stilltime => 1, timeout => 5, similarity_level => 44);
+    send_key 'delete';
+    type_string_slow_extended $test_data->{target_conf}->{id};
+    # un-check bind all IPs
+    # explicitly check use authentication only on sle15
+    # checked by default in sle12
+    if (is_sle('>=15')) {
+        wait_still_screen(stilltime => 1, timeout => 5, similarity_level => 45);
+        send_key 'alt-l';
+        wait_still_screen(stilltime => 1, timeout => 5, similarity_level => 45);
+        send_key 'alt-u';
+    }
+    wait_still_screen(stilltime => 1, timeout => 5, similarity_level => 44);
+    send_key 'alt-a';    # add LUN
+    assert_and_click('iscsi-target-LUN-path-selected', timeout => 20);
+    type_string_slow_extended $test_data->{target_conf}->{backstore};
     assert_screen 'iscsi-target-LUN';
-    send_key 'alt-o';                                                                   # OK
+    send_key 'alt-o';    # OK
     assert_screen 'iscsi-target-overview';
-    send_key 'alt-n';                                                                   # next
-    wait_still_screen(2, 10);
-    if (is_sle('<15')) {
-        send_key 'alt-a';                                                               # add client
-        send_key_until_needlematch 'iscsi-client-name-selected', 'tab';                 # there is no field shortcut, so tab till client name field is selected
-        type_string 'iqn.2016-02.de.openqa';
-        assert_screen 'iscsi-target-client-name';
-        send_key 'alt-o';                                                               # OK
-        assert_screen 'iscsi-target-client-setup';
-        send_key 'alt-n';                                                               # next
-    }
+    send_key 'alt-n';    # next
+    config_2way_authentication;
     assert_screen 'iscsi-target-overview-target-tab';
-    send_key 'alt-f';                                                                   # finish
-    wait_serial("$module_name-0", 180) || die "'yast2 iscsi-server ' didn't finish or exited with non-zero code";
-    mutex_create('iscsi_ready');                                                        # setup is done client can connect
+    send_key 'alt-f';               # finish
+    mutex_create('iscsi_ready');    # setup is done client can connect
+}
+
+sub display_targets {
+    my (%args) = @_;
+    my $cmd = 'targetcli sessions list | tee -a ' . "/dev/$serialdev";
+    assert_script_run 'targetcli ls';
```

Comment:
do you really need assert for it?

Reply:
yes, you should always get some response.